aspnetcore/security/anti-request-forgery.md

```diff
@@ -4,142 +4,230 @@ author: steve-smith
 
-ASP.NET Core anti-request-forgery default data protection configuration 
+When a user is authenticated to access an app, a server-side user session is created and stored in a database or some other persistent store. The app generates a session key that points to the stored session in the data store and includes it in the authentication ticket/cookie. The web server checks this session key each time a user requests a resource that requires authorization. The app checks whether the associated user session has sufficient privileges to access the requested resource. If the user is authorized to access the resource, the request continues. Otherwise, the request returns as unauthorized. Cookies are used to make the app appear stateful, since the app is able to maintain the user's authentication/authorization state on each request.
+
+### Token-based authentication
+
+Token-based authentication doesn't store session on the server. When a user is authenticated, they're issued a token (not an antiforgery token). The token contains user information in the form of [claims](/dotnet/framework/security/claims-based-identity-model). When a user attempts to access a resource requiring authentication, the token is sent to the app with an additional authorization header in form of Bearer token. This makes the app stateless. In each subsequent request, the token is passed in the request for server-side validation. This token isn't *encrypted*; it's *encoded*. On the server, the token is decoded to access its information. To send the token on subsequent requests, either store the token in the browser's local storage or in a cookie. Don't be concerned about CSRF vulnerability if the token is stored in the browser's local storage. CSRF is a concern when the token is stored in a cookie.
+
+### Multiple apps hosted at one domain
+
+Although `example1.cloudapp.net` and `example2.cloudapp.net` are different hosts, there's an implicit trust relationship between hosts under the `*.cloudapp.net` domain. This implicit trust relationship allows potentially untrusted hosts to affect each other's cookies (the same-origin policies that govern AJAX requests don't necessarily apply to HTTP cookies).
+
+The ASP.NET Core runtime provides some mitigation in that the username is embedded into the field token. Even if a malicious subdomain is able to overwrite a session token, it can't generate a valid field token for the user. When hosted in such an environment, the built-in anti-CSRF routines still can't defend against session hijacking or login CSRF attacks. Shared hosting environments are vulnerable to session hijacking, login CSRF, and other attacks.
+
```

Comment:
Make a note here that you can prevent this by not sharing subdomains with other people - register your own domain.